assets/app/features/upgrade/controllers/upgrade-nodes.controller.js

```diff
@@ -100,6 +150,35 @@
             vm.nodesUpgrade.totalNodes = response.data.upgraded_nodes + response.data.remaining_nodes;
             vm.nodesUpgrade.currentNodes = response.data.current_nodes;
             vm.nodesUpgrade.currentAction = response.data.current_node_action;
+            vm.nodesUpgrade.computeNodesPostponedFlag = response.data.compute_nodes_postponed;
+
+            // check for the case of postpone upgrade of compute nodes, after controller nodes were
+            // upgraded, step status would still be 'running' while substep status would be 'finished'
+            vm.nodesUpgrade.subStep = response.data.current_substep;
+            vm.nodesUpgrade.subStepStatus = response.data.current_substep_status;
+            if (vm.nodesUpgrade.subStep === 'controller_nodes' && vm.nodesUpgrade.subStepStatus === 'finished'
+                && response.data.nodes_selected_for_upgrade === 'controllers') {
+                vm.nodesUpgrade.running = false;
+
+                // show partially upgrade message
+                vm.nodesUpgrade.upgradeControllersDone = true;
+                upgradeStepsFactory.setUpgradeStep(NODE_UPGRADE_STEPS.compute);
+
+                // automatically set check box to checked after upgrade controllers
+                if (!vm.nodesUpgrade.setDefaultComputeUpgrade) {
```

Comment:
I don't see why this `setDefaultComputeUpgrade` is needed. The only code guarded with that is setting of some flags. This can be done multiple times as long as the values are the same.

Reply:
This code work similarly to line 126. Again, we don't want to reset the checkbox value in every upgrade cycle, specially if the user's choice is different that the value we're setting.

Reply:
hmmm... but `updateModel` is called only on page load and when upgrade is running. When upgrade is running user can't change these values so it's OK to sync them with the backend...

Reply:
What if the user unchecked the box and do not start the upgrade right away? For example, if he unchecks the box and gets up to get a cup of coffee. When he comes back, he'll see the box get checked back. I don't think he'll be happy about that.

Reply:
The upgrade cycle still happens after controller upgrade was done. This is because upgradeSuccess() was never called. If I read the code correctly, it's because the nodes' step status at this point still shows 'running', not 'passed'. The nodes' step status will change to passed only when ALL nodes were upgraded. That's why I have to account for the upgrade cycle. If this is not the behavior you're expecting, please let me know what the right behavior should be and how to get there.

Reply:
Ah, I never looked at the status responses in such scenario. I think we should extend the `waitForStepToEnd` function to allow for interruption of polling before success or error. I will prepare such change tomorrow.

Reply:
I created https://github.com/crowbar/crowbar-ui/pull/180
With this change you should be able to interrupt the polling when controllers are upgraded and computes upgrade should be postponed.